compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -645,11 +645,19 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
                     self.evaluate_trait_predicate_recursively(previous_stack, obligation)
                 }
 
-                ty::PredicateKind::Clause(ty::ClauseKind::HostEffect(..)) => {
-                    // FIXME(effects): It should be relatively straightforward to implement
-                    // old trait solver support for `HostEffect` bounds; or at least basic
-                    // support for them.
-                    todo!()
+                ty::PredicateKind::Clause(ty::ClauseKind::HostEffect(data)) => {
+                    self.infcx.enter_forall(bound_predicate.rebind(data), |data| {
+                        match effects::evaluate_host_effect_obligation(
```

Comment:
pretty sure that can cause undetected overflow :thinking: too tired to explain my thoughts here via comment for today, can do so in sync or maybe you're able to come up with a test yourself

Reply:
Yeah, though it's the same way that we can encounter overflow in projection afaict.

Reply:
This is just a consequence of not tracking effect goals (or projections) on the trait stack.